opentok/opentok-meet
# OPENTOK-32575: Electron app

#### What is this PR doing?

- Electron app project at `/electron`
- New npm scripts `npm run electron`, `npm run electron-build`
- Opens the `meet://` protocol in a hidden iframe for safari users which launches the app if it is installed
- Redirects safari users to `/electron/download` which links to the `.dmg` installer at https://s3.amazonaws.com/artifact.public.tokbox.com/meet/meet-electron.dmg

![screen shot 2017-02-24 at 2 47 36 pm](https://cloud.githubusercontent.com/assets/9291586/23289801/5e61c326-faa0-11e6-8420-69130b225a41.png)

#### How should this be manually tested?

1. Use the official signed build from s3 via the app
  a. Run the app locally
  b. Open a room in safari
  c. Download and install the dmg from the link
2. Test a manual launch of the app
  a. Launch the app manually
  b. Check you can publish and subscribe, and that screensharing works (screensharing will only work when https://github.com/opentok/webrtc-js/pull/2130 has merged and deployed to tbdev)
3. Test launching the app via safari
  a. Navigate to a room with safari, ensure the app launches
4. Build the app locally
  a. Do a quick run with `npm run electron`
  b. Uninstall the official build you may have installed earlier
  c. Build with `npm run electron-build`
  d. Install using the locally built dmg at `/electron/build/`
  e. Ensure you can start the locally built app

#### What are the relevant tickets?

[OPENTOK-32575](https://tokbox.atlassian.net/browse/OPENTOK-32575)
